GitUI/BuildServerIntegration/BuildServerWatcher.cs

```diff
@@ -39,6 +39,7 @@ public sealed class BuildServerWatcher : IBuildServerWatcher, IDisposable
         private readonly IRepoNameExtractor _repoNameExtractor;
         private IDisposable? _buildStatusCancellationToken;
         private IBuildServerAdapter? _buildServerAdapter;
+        private readonly object _observerLock = new object();
```

Comment:
Can be shortened in the meantime:
```suggestion
        private readonly object _observerLock = new();
```